site/hooks.ts

```diff
@@ -86,3 +98,59 @@ export const useTriggerOnEscape = (trigger: VoidFunction) => {
         }
     }, [trigger])
 }
+
+// Auto-updating Grapher bounds based on ResizeObserver
+// Optionally debounces the bounds updates
+export const useElementBounds = (
```

Comment:
I like the idea of this custom hook! Have you looked into the many existing hooks for ResizeObserver (and debouncing)? I think it might be better to use a library here to avoid the maintenance cost of doing our own thing, especially since it seems there are edge cases to take care of.

In [this example](https://usehooks-ts.com/react-hook/use-resize-observer), they use two different hooks to nicely compose behavior. Maybe we could go for a similar approach and use at least heir `seResizeObserver` here or maybe also `useDebounceCallback`?

If you think our needs are sufficiently special, I'm also fine keeping what you did, since I don't understand the problem completely.

Reply:
Good point. I opted for `usehooks-ts` now, the other hooks they provide also seem interesting and I could see us using them in the future.

I didn't use their `useDebounceCallback`, because we actually want to use a leading throttle for the best experience, imho. So I kept that as-is.

Reply:
Doesn't really matter, but just in case for the future: `useDebounceCallback` can do both leading and trailing with additional options.